app/src/main/java/com/pavesid/androidacademy/ui/details/MoviesDetailsFragment.kt

```diff
@@ -142,6 +153,37 @@ class MoviesDetailsFragment : Fragment() {
                 Status.LOADING -> binding.progressMovie.visibility = View.VISIBLE
             }
         }
+
+        animateRecycler()
+    }
+
+    private fun animateRecycler() {
+        var prev = 0.0
+        mainActivity.angle.observe(viewLifecycleOwner) {
+            val rotate = RotateAnimation(
+                prev.toFloat() % 45,
+                it.toFloat() % 45,
```

Comment:
For what you use **% 45** if in the MainActivity you limit your angle by 20?

Reply:
May be rotate views only for 5 degrees? And calculate rotation degrees like this - **rotationAngle = 5.0 * ( angleFromSensor / 20.0)** . And take out 20 to the companion objects.

Reply:
Try to rotate the items only when the angle changes more than 1.0 degrees? For more smoothness.

Reply:
45% from previous version. Forgot remove